### PR TITLE
KIF with SwiftUI

### PR DESCRIPTION
**Pull request by davideme**
ezgif com-optimize

- Add Support for SwiftUI
  - SuperHeroDetail View and ViewModel
  - SuperHeroesList View and SuperHeroesViewModel ViewModel
  - SuperHeroRow View
  - Use SDWebImageSwiftUI
  - Map LoadingView to as representable to SwiftUI
  - Update ServiceLocator 
  - Add support for Identifiable to SuperHero to be iterable for list

- Add support for catalyst, same code for iPad and Mac

- Use color in XCAssets

Cleanup:
- Remove storyboards, ViewControllers, Cells, Datasource, Delegate and presenters.

Pending next PR:
- Update Readme and Exercise branch

### CODE REVIEW

**Review by untalfranfernandez — APPROVED** [review]

File: KataSuperHeroes/SuperHeroesList.swift [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    static var previews: some View {
+        let viewModel = ServiceLocator().provideSuperHeroesViewModel()
+        viewModel.superHeroes = superHeroData
+        viewModel.isLoading = false
+        return SuperHeroesList(viewModel: viewModel)
+    }
+}
+
+
+let superHeroData = [
```
Comment (pedrovgs) [inline KataSuperHeroes/SuperHeroesList.swift]: Couldn't we reuse the hardcoded data we've declared in the repository?

File: KataSuperHeroesTests/SuperHeroesViewControllerTests.swift [hunk 2]
```diff
@@ -79,26 +70,16 @@ class SuperHeroesViewControllerTests: AcceptanceTestCase {
...
     }
 
-    func testShowsTheExactNumberOfSuperHeroes() {
-        let superHeroes = givenThereAreSomeSuperHeroes()
-
-        openSuperHeroesViewController()
-
-        let tableView = tester().waitForView(withAccessibilityLabel: "SuperHeroesTableView") as! UITableView
-        expect(tableView.numberOfRows(inSection: 0)).to(equal(superHeroes.count))
-    }
```
Comment (pedrovgs) [inline KataSuperHeroesTests/SuperHeroesViewControllerTests.swift:89]: This test was added to ensure the table view does not show more items than super heroes are. I understand now this makes no sense, but I'm wondering if there is any alternative with SwiftUI to check if a view has n items.

File: KataSuperHeroesTests/SuperHeroesViewControllerTests.swift [hunk 3]
```diff
@@ -79,26 +70,16 @@ class SuperHeroesViewControllerTests: AcceptanceTestCase {
...
         openSuperHeroesViewController()
 
-        tester().waitForView(withAccessibilityLabel: superHero.name)
-        tester().tapRow(at: IndexPath(row: superHeroIndex, section: 0),
-            inTableViewWithAccessibilityIdentifier: "SuperHeroesTableView")
+        tester().waitForView(withAccessibilityLabel: "Image of \(superHero.name)")
+        tester().tapView(withAccessibilityLabel: "Image of \(superHero.name)")
 
-        tester().waitForView(withAccessibilityLabel: superHero.name)
+        tester().waitForAbsenceOfView(withAccessibilityLabel: "Superhero name")
```
Comment (pedrovgs) [inline KataSuperHeroesTests/SuperHeroesViewControllerTests.swift:82]: Shouldn't we check the name of the superhero is being shown on the screen? I know the previous tester call wasn't correct ``tester().waitForView(withAccessibilityLabel: superHero.name)`` but the new one doesn't ensure the detail screen is open. Could we replace this one with a call to ensure the navbar shows the superhero name?

**Review by pedrovgs — CHANGES_REQUESTED** [review]

**Comment by pedrovgs** [comment]
Is there any CI check configured for this repository?

File: KataSuperHeroesTests/SuperHeroesViewControllerTests.swift (hunk 2 above)
Comment (davideme) [reply KataSuperHeroesTests/SuperHeroesViewControllerTests.swift:89]: There is not yet way to test UI elements yet. As elements like SwiftUI.List does not have properties to query them and is meant only as the definition. 
I tried using a library called [SwiftUI-Introspect](https://github.com/siteline/SwiftUI-Introspect) but the accessibility item does not match something that can be query with introspect.

File: KataSuperHeroesTests/SuperHeroesViewControllerTests.swift (hunk 3 above)
Comment (davideme) [reply KataSuperHeroesTests/SuperHeroesViewControllerTests.swift:82]: I tried doing that, but that meant a lot of changes in the code. Definitely agreed would be a better solution, but the service locator needs to be changed and AppDelegate too.

As this is a variant of KataSuperHeroesiOS, didn't wanted to introduce that many changes.

**Comment by davideme** [comment]
Set as public and added to Travis @pedrovgs

**Review by pedrovgs — APPROVED** [review]

**Pull request merged**